include/envoy/server/admin.h

```diff
@@ -153,6 +153,11 @@ class Admin {
    * @param handler the handler that will receive this Admin's listener.
    */
   virtual void addListenerToHandler(Network::ConnectionHandler* handler) PURE;
+
+  /**
+   * @return the number of worker threads to run in the server.
+   */
+  virtual uint32_t concurrency() const PURE;
```

Comment:
It feels slightly strange to hang this off of admin, but I don't feel very strongly about it. Should this just go directly on the server/factory context somewhere? wdyt?

Reply:
This was just smallest delta. It looks like there are many implementations of a factory context which all need to be updated. I am not sure what the best place to hang it off.

Reply:
I don't feel very strongly about it one way or the other. The main thing I'm concerned about is if we need to expose more options, but it's probably fine to put these here for now. @alyssawilk any thoughts on this?